torchvision/prototype/transforms/functional/_temporal.py

```diff
@@ -0,0 +1,24 @@
+import torch
+
+from torchvision.prototype import features
+
+
+def uniform_temporal_subsample(
+    inpt: features.VideoTypeJIT, num_samples: int, temporal_dim: int = -4
+) -> features.VideoTypeJIT:
+    if isinstance(inpt, torch.Tensor) and (torch.jit.is_scripting() or not isinstance(inpt, features.Video)):
+        return uniform_temporal_subsample_video(inpt, num_samples, temporal_dim=temporal_dim)
+    else:  # isinstance(inpt, features.Video)
+        if temporal_dim != -4:
+            raise ValueError("Video inputs must have temporal_dim equal to -4")
+        output = uniform_temporal_subsample_video(
+            inpt.as_subclass(torch.Tensor), num_samples, temporal_dim=temporal_dim
+        )
+        return features.Video.wrap_like(inpt, output)
+
+
+def uniform_temporal_subsample_video(video: torch.Tensor, num_samples: int, temporal_dim: int = -4) -> torch.Tensor:
+    # Reference: https://github.com/facebookresearch/pytorchvideo/blob/a0a131e/pytorchvideo/transforms/functional.py#L19
+    t_max = video.size(temporal_dim) - 1
+    indices = torch.linspace(0, t_max, num_samples, device=video.device).clamp_(0, t_max).long()
```

Comment:
Not sure if this is just for parity, but is there a reason to create the linspace in `int32` to later convert it to `int64`?

Reply:
Plus, maybe this is just personal preference, but I would appreciate us to be "explicit" about the types

```suggestion
    indices = torch.linspace(0, t_max, num_samples, device=video.device).clamp_(0, t_max).to(torch.int64)
```

Reply:
Plusplus, the docs for [`torch.linspace`](https://pytorch.org/docs/stable/generated/torch.linspace.html) state:

> Creates a one-dimensional tensor of size `steps` whose values are evenly spaced from `start` to `end`, inclusive.

What is the clamp for if the function by definition does not return values outside this range?

Reply:
Let's be careful on how we are porting this. Here is why casting later is not the same:
```python
t_max = 9
num_samples = 8
indices = torch.linspace(0, t_max, num_samples, dtype=torch.int64)
indices2 = torch.linspace(0, t_max, num_samples).clamp(0, t_max).long()
assert indices.equal(indices2), f"({t_max}, {num_samples})\n{indices}\n{indices2}"
```
Result:
```
    assert indices.equal(indices2), f"({t_max}, {num_samples})\n{indices}\n{indices2}"
AssertionError: (9, 8)
tensor([0, 1, 2, 3, 5, 6, 7, 8])
tensor([0, 1, 2, 3, 5, 6, 7, 9])
```

Reply:
IMO, this looks like a bug in torch linspace as its docs states that the last element is `end` (=t_max)